esp-wifi/src/timer/riscv.rs

```diff
@@ -28,8 +28,17 @@ const TIMER_DELAY: fugit::HertzU32 = fugit::HertzU32::from_raw(crate::CONFIG.tic
 static ALARM0: Mutex<RefCell<Option<Alarm<Periodic, 0>>>> = Mutex::new(RefCell::new(None));
 
 pub fn setup_timer(systimer: TimeBase) {
+    unsafe {
+        riscv::interrupt::disable();
+    }
+
     let alarm0 = systimer.into_periodic();
     alarm0.set_period(TIMER_DELAY.into());
+    // make sure the period is loaded and reset - should be done in esp-hal ultimately
```

Comment:
How hard is the fix in esp-hal, why don't we try and fix it there before the next release?

Reply:
Probably because Jesse said the next release is today :D

Reply:
If we can get a PR merged in the next few hours then we can include it in today's releases. Otherwise will have to wait until next round.